src/ui/components/forms/VPNComboBox.qml

```diff
@@ -127,4 +130,71 @@ ComboBox {
 
         }
     }
+
+    states: [
+        State {
+            name: "empty"
+            when: combo.currentIndex === -1
+                  && !combo.hovered
+                  && !comboPopup.visible
+
+            PropertyChanges {
+                target: combo.contentItem
+                color: Color.input.default.placeholder
+            }
+
+            PropertyChanges {
+                target: combo.background
+                border.color: Color.input.default.border
+                border.width: 1
+            }
+        },
+        State {
+            name: "emptyHovered"
+            when: combo.hovered
+                  && combo.currentIndex === -1
+                  && !comboPopup.visible
+
+            PropertyChanges {
+                target: combo.contentItem
+                color: Color.input.hover.text
+            }
+
+            PropertyChanges {
+                target: combo.background
+                border.color: Color.input.hover.border
+                border.width: 1
+            }
+        },
+        State {
```

Comment:
Let's move this "focused" state block to the top. Right now if you tab to combo box and you haven't already selected an option, the conditionals for "empty" will evaluate to true and the focus property changes don't get set. Otherwise this looks awesome.